packages/core-commands/src/site-editor-navigation-commands.js

```diff
@@ -177,8 +201,8 @@ const getNavigationCommandLoaderPerTemplate = ( templateType ) =>
 
 		const commands = useMemo( () => {
 			if (
-				! isBlockBasedTheme &&
-				! templateType === 'wp_template_part'
+				! canCreateTemplate ||
+				( ! isBlockBasedTheme && ! templateType === 'wp_template_part' )
```

Comment:
I actually wonder if this second condition is needed.

Reply:
```
( ! isBlockBasedTheme && ! templateType === 'wp_template_part' )
```

This may be an edge case, but it's here to prevent themes like the one below from unintentionally accessing the site editor.

- The theme has template part files (e.g. `parts/header.html`)
- But the theme is not opted in to `add_theme_support( 'block-template-parts' )`

If a theme does not support `block-template-parts`, the theme will [not have access to the site editor](https://github.com/WordPress/wordpress-develop/blob/f003fd07ba5da8dd0266715d3874250040eab605/src/wp-admin/site-editor.php#L30) and should not expose template parts in the commands.